src/integrationTest/java/uk/gov/hmcts/reform/juddata/cameltest/testsupport/JrdBatchIntegrationSupport.java

```diff
@@ -96,4 +112,70 @@
 
     @Value("${truncate-audit}")
     protected String truncateAudit;
+
+    public void validateDbRecordCountFor(String queryName, int expectedCount) {
+        assertEquals(expectedCount, jdbcTemplate.queryForList(queryName).size());
+    }
+
+    public void validateExceptionDbRecordCount(String queryName, int expectedCount, boolean isPartialSuccessValidation) {
+        List<Map<String, Object>> exceptionList = jdbcTemplate.queryForList(queryName);
+
+        exceptionList.forEach(exception -> {
+            assertNotNull(exception.get("scheduler_name"));
+            assertNotNull(exception.get("scheduler_start_time"));
+            assertNotNull(exception.get("error_description"));
+            assertNotNull(exception.get("updated_timestamp"));
+            if (isPartialSuccessValidation) {
+                assertNotNull(exception.get("table_name"));
+                assertNotNull(exception.get("key"));
+                assertNotNull(exception.get("field_in_error"));
+            }
+        });
+        assertEquals(expectedCount, exceptionList.size());
+    }
+
+    public void validateUserProfileFile() {
+        List<Map<String, Object>> judicialUserProfileList = jdbcTemplate.queryForList(userProfileSql);
+        assertEquals(judicialUserProfileList.size(), 2);
+        Contracts.assertNotNull(judicialUserProfileList.get(0));
+        Contracts.assertNotNull(judicialUserProfileList.get(1));
+        assertEquals(judicialUserProfileList.get(0).get("elinks_id"), "1");
+        assertEquals(judicialUserProfileList.get(1).get("elinks_id"), "2");
+        assertEquals(judicialUserProfileList.get(0).get("email_id"), "joe.bloggs@ejudiciary.net");
+        assertEquals(judicialUserProfileList.get(1).get("email_id"), "jo1e.bloggs@ejudiciary.net");
+    }
+
```

Comment:
Move these methods in ParentIntegrationTestSupport

Reply:
Done